tests/unit_tests/test_retry.py

```diff
@@ -0,0 +1,87 @@
+"""retry tests"""
+
+from unittest import mock
+
+import pytest
+
+from wandb.sdk.lib import retry
+
+
+def noop_sleep(_):
+    pass
+
+
+def test_retry_respects_num_retries():
+    func = mock.Mock()
+    func.side_effect = ValueError
+
+    retrier = retry.Retry(
+        func,
+        num_retries=7,
```

Comment:
total nit: do you think it makes sense to change `7` and values in other tests to variables. i.e. something like this
```
num_retires = 7
retrier = retry.Retry(..., num_retires=num_retires)
assert func.call_count == num_retries + 1
```
totally up to you... i don't have a strong preference either way